modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/incoming/IncomingSnapshotCopier.java

```diff
@@ -113,8 +147,174 @@ public SnapshotReader getReader() {
         return new IncomingSnapshotReader(snapshotMeta);
     }
 
-    @Override
-    public void close() throws IOException {
-        threadPool.shutdownNow();
+    private void start0() {
+        if (!cancelBusyLock.enterBusy()) {
+            return;
+        }
+
+        try {
+            partitionSnapshotStorage.partition().reCreatePartition()
+                    .whenComplete((unused, throwable) -> {
+                        if (throwable != null) {
+                            future.completeExceptionally(copierException(throwable, RaftError.EIO, "Error while recreating partition"));
+                        } else {
+                            ClusterNode snapshotSenderNode = partitionSnapshotStorage.topologyService()
+                                    .getByConsistentId(snapshotUri.nodeName);
+
+                            if (snapshotSenderNode == null) {
+                                future.completeExceptionally(copierException(
+                                        new IgniteException("Sender node was not found or it is offline: " + snapshotUri.nodeName),
+                                        RaftError.UNKNOWN,
+                                        "Sender node was not found or it is offline"
+                                ));
+                            } else {
+                                requestSnapshotMetaAsync(snapshotSenderNode);
+                            }
+                        }
+                    });
+        } catch (StorageException e) {
+            future.completeExceptionally(copierException(e, RaftError.EIO, "Error recreating partition"));
+        } catch (Throwable t) {
+            future.completeExceptionally(t);
+        } finally {
+            cancelBusyLock.leaveBusy();
+        }
+    }
+
+    private void requestSnapshotMetaAsync(ClusterNode snapshotSenderNode) {
+        if (!cancelBusyLock.enterBusy()) {
+            return;
+        }
+
+        try {
+            partitionSnapshotStorage.outgoingSnapshotsManager().messagingService().invoke(
+                    snapshotSenderNode,
+                    MSG_FACTORY.snapshotMetaRequest().id(snapshotUri.snapshotId).build(),
+                    NETWORK_TIMEOUT
+            ).whenComplete((networkMessage, throwable) -> {
```

Comment:
you can call it `response`, would be easier to understand

Reply:
Tried to fix it.